### ISSUE TITLE
Add `crypt(3)` 

### ISSUE THREAD

**Issue by SteveLauC** [rust-lang/libc#2880]
It would be great if we have `crypt(3)` exposed in `libc`. On `openbsd/freebsd`, it is marked as deprecated so I am not that sure it should be added.

##### doc
* [linux](https://man7.org/linux/man-pages/man3/crypt.3.html)
* [netbsd](https://man.netbsd.org/crypt.3)
* [dragonfly](https://man.dragonflybsd.org/?command=crypt&section=3)

**Comment by lopopolo** [comment]
I would also love for this to be exposed in `libc` to aid in the implementation of `String#crypt` in Artichoke Ruby.

My preference would for the bindings to be available on all platforms that support it.

**Comment by SteveLauC** [comment]
Looks like the following code snippet will work:
```rust
#[link(name= "crypt",)]
extern "C" {
    pub fn crypt(phrase: *const c_char, setting: *const c_char) -> *mut c_char;
}
```

I tested it on my Linux machine:
```rust
fn main() {
    let str: [u8; 7] = [b'H', b'E', b'L', b'L', b'O', b'!', 0];
    let setting: [u8; 4] = [b'$', b'1', b'$', 0];
    let format_str: [u8; 4] = [b'%', b's', b'\n', 0];

    unsafe {
        let res = crypt(str.as_ptr() as *mut i8, setting.as_ptr() as *mut i8);
        printf(format_str.as_ptr() as *mut i8, res);
    }
}
```
```shell
$ cargo r -q
$1$$ar9Qb9TEF.hJqAF0vp/Os1
```

**Comment by piegamesde** [comment]
I don't think adding a deprecated API is a good idea. It's a good occasion to switch to alternatives to libxcrypt.

**Comment by lopopolo** [comment]
I don't believe libc bindings are the place to make value judgements on what APIs should or should not be used.

If we feel strongly about it, I'd be open to having the bindings marked with a `#[deprecated]` annotation, but I think the bindings themselves should exist.

I trust the `libc` crate maintainers to do a better job of declaring the bindings correctly on all platforms than I could.

**Comment by workingjubilee** [comment]
If you inspect the source of this crate, you will notice that when maintainers of targets express that APIs should be deprecated, directly to us, we tend to indulge them. The *BSD deprecation periods do last years, but they _are_ willing to remove things they've deprecated, as they are quite serious about that. I don't believe it is appropriate to dare FreeBSD or OpenBSD to break our libc bindings on their platform with their next release.

Besides, the reentrant version is `crypt_r`.
